Library/RSBot.Core/Components/PickupManager.cs

```diff
@@ -116,63 +116,67 @@ public static void RunPlayer(Position playerPosition, Position centerPosition, i
             }
         }
 
-        public static void RunAbilityPet( Position centerPosition, int radius = 50 ) 
+        public static void RunAbilityPet(Position centerPosition, int radius = 50)
         {
-            if( RunningAbilityPetPickup || !UseAbilityPet || !Game.Player.HasActiveAbilityPet )
+            if (RunningAbilityPetPickup || !UseAbilityPet || !Game.Player.HasActiveAbilityPet)
                 return;
 
             RunningAbilityPetPickup = true;
-            try {
-                if( !SpawnManager.TryGetEntities<SpawnedItem>( out var entities, ( si ) => Condition( si, centerPosition, radius ) 
-                    && PickupFilter.Any( p => p.CodeName == si.Record.CodeName && !p.PickOnlyChar ) ) ) 
+            
+            try
+            {
+                if (!SpawnManager.TryGetEntities<SpawnedItem>(out var entities, (si) => Condition(si, centerPosition, radius)
+                    && (si.Record.IsGold || PickupFilter.Any(p => p.CodeName == si.Record.CodeName && !p.PickOnlyChar))))
```

Comment:
Gold isn't a registered item in the pickup filter. But since the Condition()-Method does only return gold if it's configured, we're fine with the "record.IsGold" check only. It would not appear if the previous condition failed.

Reply:
ye you are right.